frb_dart/lib/src/ffi.dart

```diff
@@ -1,3 +1,74 @@
+import 'package:meta/meta.dart';
+import 'ffi/io.dart' if (dart.library.html) 'ffi/web.dart';
+
 export 'ffi/stub.dart'
     if (dart.library.io) 'ffi/io.dart'
     if (dart.library.html) 'ffi/web.dart';
+
+typedef DropFnType = void Function(PlatformPointer);
+typedef ShareFnType = PlatformPointer Function(PlatformPointer);
+
+/// An opaque pointer to a native C or Rust type.
+/// Recipients of this type should call [dispose] at least once during runtime.
+/// If passed to a native function after being [dispose]d, an exception will be thrown.
+abstract class FrbOpaque extends FrbOpaqueImpl {
```

Comment:
uber-nit: `Sth extends SthImpl` looks a bit weird. what about renaming "impl" to something like "base"